### ISSUE TITLE
Alpha version for ActivationResult (data storage & retrieval)

### ISSUE THREAD

**Issue by fohrloop** [fohrloop/wakepy#52]
## Background

The modes in wakepy current 0.8.0dev return a `ActivationResult` object, like this:

```python
with keep.running() as k:
    if not k.success:
        print('failed to set keepawake')
```

here `k` is an `ActivationResult` object.

## Task

Edit: Marking Communication with the thread which switches the mode out of scope. In scope: Just the data storage and retrieval mechanism for ActivationResult

- Make constructing ActivationResult objects work.
- Properties:
  -  success
  - failure
  - real_success
- Methods:
  - get_details()
- Create unit tests for it.

**Comment by fohrloop** [comment]
I've been thinking about the API for the ActivationResult objects. It would be nice to be able to know at least
- When setting keepawake was not successful: which methods were tried, and what were the failure reasons.
- When setting keepawake was successful: which method(s) were used (just names are enough)


Here are some initial thoughts. The API could look something like this:

```python
with keep.running() as k:

    if not k.success:
        logger.warning('keepawake not successful')
        logger.warning(k.get_details())
        
    else: 
       logger.info('keepawake successful. Using %s', k.active_methods)
       # do something
```

### ActivationResult.get_details()

By default, `k.get_details()` should return details from "requirements" and "activation" stages. The platform check stage is omitted since it is usually not interesting to know about methods that failed since the platform is not supported (the list might be long).

The `k.get_details` should return something like:

```python
[
  (FAIL, REQUIREMENTS, 'method', 'Requires Xfce v.x.y.z'),
  (FAIL, ACTIVATION 'other-method', 'ValueError: Foo'),
  (SUCCESS, REQUIREMENTS, 'some.Third.Method', ''),
  (UNUSED, NONE, 'FourthMethod', ''),
]
```

Each entry should have format

```
(SuccessStatus, StageName, Method.name, message)
```

where
- SuccesStatus is FAIL, SUCCESS or UNUSED
- StageName is one of the stages in the activation process. In order: PLATFORM_SUPPORT, REQUIREMENTS, ACTIVATION - or NONE
- Method.name is the name from the method class
- message is an optional message from the stage output, and should contain a helpful error message if coming from REQUIREMENTS or ACTIVATION stage.

The entries in the return value of `get_details()` should be in their priority order. Optional parameters for `get_details()`:

```
platform_stage: bool
   Controls if items from platform support check stage should be included. Default: False
include_success: bool
   If True, will include the methods which were successful. Default: True.
include_fail: bool
   If True, will include the methods which were failing. Default: True.
include_remaining: bool
   If True, will include the methods which were not used at all; Those with SuccesStatus of UNUSED. Default: True
```

### ActivationResult.active_methods

This should be the easy way of letting user know which methods were used. This is a list of strings, like 
```
['some-method', 'org.gnome.SessionManager:Inhibit', 'YetAnotherMethod']
```
The maximum length of the list is `max_methods`. Might be useful to have also something like `k.active_methods_as_string` which would be in this case the string literal `'some-method, org.gnome.SessionManager:Inhibit & YetAnotherMethod'`. (note: only allow whitelisted characters in method names. Disallowed: ` `(space), `,` (comma), `'` (apostrophe) and `"` (quotation mark). )

**Comment by fohrloop** [comment]
Could be that the higher level class should be called `ModeActivationResult` (given to user), and then the `ModeActivationResult.get_details()` should return a list of `MethodActivationResult` objects. Returning a list of tuples would mean that the API is not easily extendable.

**Comment by fohrloop** [comment]
Made the scope of this task a bit smaller; Only consider about the data storage and retrieval part of the ActivationResult. The communication with the thread will be its own ticket.

**Comment by fohrloop** [comment]
Changes merged to `main` in PR #57
